contracts/FrendLend.sol

```diff
@@ -0,0 +1,131 @@
+    uint256 public fee;
+    /// the number of loan offers
+    uint256 public loanOfferCount;
+    /// a mapping of id to the FinanceTerms offered by the creditor
+    mapping(uint256 => LoanOffer) public loanOffers;
+
+    event LoanOffered(uint256 indexed loanId, address indexed offerredBy, LoanOffer loanOffer, uint256 blocktime);
+    event LoanOfferAccepted(uint256 indexed loanId, uint256 indexed claimId, uint256 blocktime);
+    event LoanOfferRejected(uint256 indexed loanId, address indexed rejectedBy, uint256 blocktime);
+    event BullaTagUpdated(address indexed bullaManager, uint256 indexed tokenId, address indexed updatedBy, bytes32 tag, uint256 blocktime);
+
+    error INSUFFICIENT_FEE();
+    error NOT_CREDITOR();
+    error NOT_DEBTOR();
+    error NOT_CREDITOR_OR_DEBTOR();
+    error NOT_ADMIN();
+    error INVALID_TERM_LENGTH();
+    error WITHDRAWAL_FAILED();
+    error TRANSFER_FAILED();
+
+    constructor(
+        IBullaClaim _bullaClaim,
+        address _admin,
+        uint256 _fee
+    ) {
+        bullaClaim = _bullaClaim;
+        admin = _admin;
+        fee = _fee;
+    }
+
+    ////// ADMIN FUNCTIONS //////
+
+    /// @notice SPEC:
+    ///     allows an admin to withdraw `withdrawableFee` amount of tokens from this contract's balance
+    ///     Given the following: `msg.sender == admin`
+    function withdrawFee(uint256 _amount) public {
+        if (msg.sender != admin) revert NOT_ADMIN();
+
+        (bool success, ) = admin.call{ value: _amount }('');
+        if (!success) revert WITHDRAWAL_FAILED();
+    }
+
+    ////// USER FUNCTIONS //////
+
+    function offerLoan(LoanOffer calldata offer) public payable {
+        if (msg.value != fee) revert INSUFFICIENT_FEE();
+        if (msg.sender != offer.creditor) revert NOT_CREDITOR();
+        if (offer.termLength == 0) revert INVALID_TERM_LENGTH();
+
+        uint256 offerCount = ++loanOfferCount;
+        loanOffers[offerCount] = offer;
+
+        emit LoanOffered(offerCount, msg.sender, offer, block.timestamp);
+    }
+
+    function rejectLoanOffer(uint256 loanId) public {
+        LoanOffer memory offer = loanOffers[loanId];
+        if (msg.sender != offer.creditor || msg.sender != offer.debtor) revert NOT_CREDITOR_OR_DEBTOR();
+
+        delete loanOffers[loanId];
+
+        emit LoanOfferRejected(loanId, msg.sender, block.timestamp);
+    }
+
+    // @dev NOTE: does not accept fee on transfer tokens
+    function acceptLoan(
+        uint256 loanId,
+        string calldata tokenURI,
+        bytes32 tag
+    ) public {
+        LoanOffer memory offer = loanOffers[loanId];
+        if (msg.sender != offer.debtor) revert NOT_DEBTOR();
+
+        delete loanOffers[loanId];
+
+        uint256 claimAmount = offer.loanAmount + (offer.loanAmount * offer.interestBPS) / MAX_BPS;
+        uint256 claimId = bullaClaim.createClaimWithURI(
+            offer.creditor,
+            offer.debtor,
+            offer.description,
+            claimAmount,
```

Comment:
should we add + 1 here too?

Reply:
and maybe so UI shenanigans to hide this mecanism

Reply:
Hmm. I think i see your point, the creditor pays 1 wei + the loan amount, the debtor only has to pay back loan amount + total interest.
I don't have a preference for this POC, what do you think?